### PR TITLE
Libstring fs

### PR DESCRIPTION
**Pull request by Athinanarof**
## What is the problem/goal being addressed?
LibString.fs functions implemented. Unit test included.
## What is the solution to this problem?
Functions implemented.
## What are the changes here? How do they solve the problem and what other product impacts do they cause?
*Please include before/after screenshots/gifs if appropriate*

## How are you sure this works/how was this tested?
Unit test included for each function. However, I have 3 test cases of slice_v0 to continue working on. WIP.

## What is the reversion plan if this fails after shipping?
Revert changes.

### CODE REVIEW

**Comment by Athinanarof** [comment]
Hi @pbiggar I have created a new MR.

File: fsharp-backend/src/LibExecution/StdLib/LibString.fs [hunk 1]
```diff
@@ -883,8 +883,25 @@ let fns : List<BuiltInFn> =
       fn =
         (function
         | _, [ DStr s; DInt n ] ->
-            let characterCount = int n
-            Value(DStr(s.[characterCount..]))
+            let drop_first_n s (num_egcs : bigint) =
```
Comment (pbiggar) [inline fsharp-backend/src/LibExecution/StdLib/LibString.fs:1000]: This looks a bit complex. I think you could have the same effect by doing:

```
s
|> String.toEgcSeq
|> Seq.toList
|> List.skip count
|> String.concat ""

Comment (Athinanarof) [reply fsharp-backend/src/LibExecution/StdLib/LibString.fs:1000]: Hi @pbiggar, I know there were simpler ways to make some functions work but what I did was to translate the current implementation in Ocalm, like in this one uses Unicode_string.drop_first_, I thought I should be doing that 😅
image

**Review by pbiggar — COMMENTED** [review]
I was going to suggest changing all the behaviour to more idiomatic functional code, but I see your point, so I guess it's better to stick with direct translations.

Can you:
- make sure there are complex emoji tests for everything
- change the code to use camelCase
- use `0I` or `1I` instead of `bigint0` or `bigint`

And I guess we'll ship like that.

Thanks!

**Comment by Athinanarof** [comment]
Hi @pbiggar, changes are done; I just found two functions where the complex emojis do not work; I will continue with the file in the morning, and I will add those in the next MR.

**Comment by pbiggar** [comment]
Great, thanks! Here's a bunch more challenging unicode characters to try: https://stackoverflow.com/a/51539774/104021

**Pull request merged**